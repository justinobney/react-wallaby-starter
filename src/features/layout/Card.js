/** @jsx createElement */
import {createElement} from 'react';
import {Container, Header} from 'semantic-ui-react';
import glamorous from 'glamorous';

import {colors, contentBase} from 'styles';

const Wrapper = glamorous(Container)(contentBase);
const Title = glamorous(Header)({
  color: `${colors.brandPrimary} !important`,
});

const Card = ({children, title, ...props}) => (
  <Wrapper {...props}>
    {!!title && <Title as="h1">{title}</Title>}
    {children}
  </Wrapper>
);

export default Card;
