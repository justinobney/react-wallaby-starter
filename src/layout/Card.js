/** @jsx createElement */
import {createElement} from 'react';
import {Container, Header} from 'semantic-ui-react';
import glamorous from 'glamorous';
import {matcher, serializer} from 'jest-glamor-react';

import {colors, contentBase} from 'styles';

expect.addSnapshotSerializer(serializer);
expect.extend(matcher);

const Wrapper = glamorous(Container)(contentBase);
const Title = glamorous(Header)({
  color: `${colors.brandPrimary} !important`,
});

const Card = ({children, title, ...props}) => (
  <Wrapper {...props}>
    {!!title && <Title as="h1">{title}</Title>}
    {children}
  </Wrapper>
);

export default Card;
