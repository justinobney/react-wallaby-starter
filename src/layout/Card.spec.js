import Card from './Card';
import React from 'react';
import {create} from 'react-test-renderer';

describe('Component: Card', () => {
  it('should render with title and children', () => {
    const tree = create(<Card title="Title" children="Content" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should allow skipping title', () => {
    const tree = create(<Card children="Content" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
