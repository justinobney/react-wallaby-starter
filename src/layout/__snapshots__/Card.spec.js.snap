// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Card should allow skipping title 1`] = `
.css-j7k7y6,
[data-css-j7k7y6] {
  box-shadow: 0 0 2px rgba(0, 0, 0, 0.5);
  margin: 0 auto 20px;
  padding: 20px;
  border: solid 1px rgba(0, 0, 0, 0.2);
  background: #FFF;
}

<div
  className="ui container css-j7k7y6"
>
  Content
</div>
`;

exports[`Component: Card should render with title and children 1`] = `
.css-j7k7y6,
[data-css-j7k7y6] {
  box-shadow: 0 0 2px rgba(0, 0, 0, 0.5);
  margin: 0 auto 20px;
  padding: 20px;
  border: solid 1px rgba(0, 0, 0, 0.2);
  background: #FFF;
}

.css-y2vsc,
[data-css-y2vsc] {
  color: #00573d !important;
}

<div
  className="ui container css-j7k7y6"
>
  <h1
    className="ui header css-y2vsc"
  >
    Title
  </h1>
  Content
</div>
`;
